### PR TITLE
Add a modal to allow the edit of an approval process sequence

### PR DESCRIPTION
**Pull request by lgalis**
First draft for sequence editing - a menu item and a modal to set the sequence.

Screenshot from 2020-02-11 13-04-20

Screenshot from 2020-02-11 16-15-16

### CODE REVIEW

File: src/smart-components/workflow/edit-workflow-info-modal.js [hunk 1]
```diff
@@ -29,35 +31,48 @@ const EditWorkflowInfoModal = ({
...
+      title: `Edit approval process`,
       dismissable: true,
-      description: `Edit approval process's groups was cancelled by the user.`
+      description: `Edit approval process was cancelled by the user.`
     });
     push('/workflows');
   };
 
+  const formValid = () => {
+    console.log('Debug - formData, valid', formData, (editType === 'sequence' ?
```
Comment (Hyperkid123) [inline src/smart-components/workflow/edit-workflow-info-modal.js]: Console log

File: src/smart-components/workflow/edit-workflow-info-modal.js [hunk 2]
```diff
@@ -69,7 +84,7 @@ const EditWorkflowInfoModal = ({
                   id="save-edit-workflow-info"
                   variant="primary"
                   type="submit"
-                  isDisabled={ isFetching }
+                  isDisabled={ isFetching || !formValid(formData) }
```
Comment (Hyperkid123) [inline src/smart-components/workflow/edit-workflow-info-modal.js]: Pf4 forms should have the submit button enabled even if there are validation errors. If that is the case, cliking on the submit should focus the first invalid field

**Comment by lgalis** [comment]
@Hyperkid123 - feedback addressed

**Comment by sbuenafe-rh** [comment]
@lgalis sorry, but I noticed that for editing the sequence number, the modal's title should be **Edit sequence**

For the field label, **Enter sequence**, is the admin allow to type in the number and/or allow to use the dropdown to choose the number? What happens if the admin chooses a sequence number that is already in use? 

Also, was the sequence column suppose to be the left most column?

**Comment by lgalis** [comment]
@sbuenafe-rh - the admin can enter a sequence. The entered sequence can be one that exosst already, in fact this will be the main purpose - to move a workflow up or down in priority. The workflow will be assigned the sequence entered and all the workflows with higher sequences are automatically adjusted by the API. I will fix the Title and field label. Let me know if you want the sequence column moved.

**Comment by mkanoor** [comment]
@lgalis How would the user know what is the highest index to enter, if they like to move it to the end. They would have to guess different numbers, can there be specific values like "top", "bottom" that the API could mean put the workflow at the top/bottom without knowing how many there are.

**Pull request merged**